### PR TITLE
Handle local paths in Hugging Face file probe

### LINKED ISSUE #1034: probes: fileformats probe fails when targeting a local modal path
The `fileformats` are limited to huggingface generators, however there are cases where the generator can load a local checkpoint that would not be complatible with this probe. This is due the the expectation that the model under test is published for file enumeration via the `huggingface_hub` utility library.

Contrived example:
```
python -m garak -m huffingface.Model -n /usr/local/checkpoints/Mistral-NeMo-Minitron-8B-Instruct -p fileformats
```

Current trace reported in logs:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/garak/cli.py", line 530, in main
    command.probewise_run(
  File "/usr/local/lib/python3.10/site-packages/garak/command.py", line 237, in probewise_run
    probewise_h.run(generator, probe_names, evaluator, buffs)
  File "/usr/local/lib/python3.10/site-packages/garak/harnesses/probewise.py", line 107, in run
    h.run(model, [probe], detectors, evaluator, announce_probe=False)
  File "/usr/local/lib/python3.10/site-packages/garak/harnesses/base.py", line 114, in run
    attempt_results = probe.probe(model)
  File "/usr/local/lib/python3.10/site-packages/garak/probes/fileformats.py", line 63, in probe
    repo_filenames = huggingface_hub.list_repo_files(generator.name)
  File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 106, in _inner_fn
    validate_repo_id(arg_value)
  File "/usr/local/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 154, in validate_repo_id
    raise HFValidationError(
huggingface_hub.errors.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/usr/local/checkpoints/Mistral-NeMo-Minitron-8B-Instruct/'. Use `repo_type` argument if needed.
```

This probe needs to be able to either skip for `local` model paths or follow an alternative file gathering approach.

### PR DESCRIPTION
**Pull request by musaabhasan**
## Summary
- Detect when a Hugging Face generator name points to a local filesystem path and gather files from that path directly instead of calling Hugging Face Hub.
- Skip empty local directories cleanly so the fileformats probe does not create an empty `Attempt.outputs` update.
- Catch `HFValidationError` from ambiguous non-Hub names and skip the probe instead of letting the validation error escape.
- Catch `OfflineModeIsEnabled` when Hugging Face Hub offline mode is enabled and skip the probe with a warning.
- Add regression coverage proving local paths do not call `huggingface_hub.list_repo_files()` and offline Hub mode is handled cleanly.

Fixes #1034.

## Testing
- `.venv\Scripts\python.exe -m pytest tests\probes\test_probes_fileformats.py -k "hf_files_load or local_directory or empty_local_directory or offline_mode" -q` - 4 passed, 1 deselected
- `.venv\Scripts\python.exe -m compileall garak\probes\fileformats.py tests\probes\test_probes_fileformats.py`
- `git diff --check` - no whitespace errors; local Git emitted Windows autocrlf warnings

Full `tests/probes/test_probes_fileformats.py` was not completed in this local venv because the pre-existing `test_hf_files_hf_repo` imports `garak.generators.huggingface`, which requires `torch` and this lightweight test environment does not include it.

### CODE REVIEW

**Comment by musaabhasan** [comment]
I have read the DCO Document and I hereby sign the DCO

**Comment by musaabhasan** [comment]
recheck

File: garak/probes/fileformats.py [hunk 1]
```diff
@@ -65,18 +87,39 @@ def probe(self, generator) -> Iterable[garak.attempt.Attempt]:
...
-            desc=f"Gathering files in {generator.name}",
-            colour=f"#{garak.resources.theme.PROBE_RGB}",
-        ):
-            local_filename = huggingface_hub.hf_hub_download(
-                generator.name, repo_filename, force_download=False
+        local_filenames = self._get_local_files(generator.name)
+        if local_filenames is None:
+            try:
+                repo_filenames = huggingface_hub.list_repo_files(generator.name)
+            except HFValidationError:
```
Comment (jmartin-tech) [inline garak/probes/fileformats.py:99]: Testing shows this can also raise `OfflineModeIsEnabled` when the `HF_HUB_OFFLINE` environment variable is set, presenting another possible edge case.
```suggestion
            except huggingface_hub.errors.OfflineModeIsEnabled as e:
                logging.warning("File enumeration is not available when offline mode is enabled.")
                return []
            except HFValidationError:
```

**Review by jmartin-tech — CHANGES_REQUESTED** [review]
This works for the targeted issue, sorry to expand the scope, testing shows an additional edge case that could use a guard as well.

Please be sure to validate suggestions and consider adding an additional test for the additional edge case.

The behavior when the no files are found leaves something to be desired, the report simply omits the probe silently since no attempts generated. That may be something the project should address separately however.

**Comment by musaabhasan** [comment]
Thanks for the catch. I added a guard for `OfflineModeIsEnabled` before the `HFValidationError` fallback and included a regression test that simulates Hugging Face Hub offline mode.

Validation run locally:

- `.venv\Scripts\python.exe -m pytest tests\probes\test_probes_fileformats.py -k "hf_files_load or local_directory or empty_local_directory or offline_mode" -q` - 4 passed, 1 deselected
- `.venv\Scripts\python.exe -m compileall garak\probes\fileformats.py tests\probes\test_probes_fileformats.py`
- `git diff --check` - no whitespace errors; Windows reported the existing autocrlf normalization warning

**Review by jmartin-tech — APPROVED** [review]
